osf/models/node.py

```diff
@@ -2409,13 +2409,44 @@ def _add_related_source_tags(self, contributor):
         ).first() or osf_provider_tag
         contributor.add_system_tag(source_tag)
 
-    def _remove_from_associated_collections(self):
+    def _remove_from_associated_collections(self, auth=None, force=False):
         for submission in self.guids.first().collectionsubmission_set.all():
             associated_collection = submission.collection
             if associated_collection.is_bookmark_collection and not self.deleted:
                 if self.contributors.filter(pk=associated_collection.creator.id).exists():
                     continue
-            submission.delete()
+
+            if submission.state not in [CollectionSubmissionStates.REMOVED, CollectionSubmissionStates.REJECTED]:
+                user = getattr(auth, 'user', None)
+
+                if submission.state == CollectionSubmissionStates.ACCEPTED:
+                    submission.remove(
+                        user=user,
+                        comment='Removed from collection due to implicit removal due to privacy changes.',
+                        removed_due_to_privacy=True
+                    )
+                elif submission.state == CollectionSubmissionStates.PENDING and user:
+                    submission.reject(
+                        user=user,
+                        comment='Rejected from collection due to implicit removal due to privacy changes.',
+                        force=True
+                    )
+                elif force and submission.state == CollectionSubmissionStates.ACCEPTED:
+                    request, user_id = get_request_and_user_id()
+                    submission.remove(
+                        user=request.user,
+                        comment='Removed from collection via system command.',  # typically spam
+                        force=True
+                    )
+                elif force and submission.state == CollectionSubmissionStates.PENDING:
+                    request, user_id = get_request_and_user_id()
+                    submission.rejected(
```

Comment:
`rejected` -> `reject`?

Reply:
Gotta write some test for confirming this as spam.